src/apps/api/views/analytics.py

```diff
@@ -0,0 +1,115 @@
+from django.db.models import Count, F
+from django.contrib.auth import get_user_model
+from rest_framework.decorators import api_view, renderer_classes
+from rest_framework.response import Response
+from rest_framework.renderers import JSONRenderer
+from rest_framework_csv import renderers as r
+from competitions.models import Competition, Submission
+
+
+User = get_user_model()
+
+
+def build_request_object(model_name, filter_param, time_unit, start_date, end_date, csv):
+    filter_args = {
+        filter_param + '__range': [start_date, end_date]
+    }
+
+    data = model_name.objects.filter(**filter_args).dates(filter_param, time_unit).annotate(count=Count('pk')).values('count', x=F('datefield'))
```

Comment:
looks like you can put the `annotate` directly in the `values`, cool.
Also, let's use a more descriptive variable name than `x`. Maybe something like this:
```python
data = model_name.objects.filter(**filter_args).dates(filter_param, time_unit).values(count=Count('pk'), _datefield=F('datefield'))
```

Reply:
Great catch. Done.